configs/10.0/dockerfile-devel

```diff
@@ -12,37 +12,40 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG UBUNTU_RELEASE=xenial
+ARG ARCH
+ARG DISTRO_NAME
+ARG DISTRO_NICK
 
-FROM ppc64le/ubuntu:${UBUNTU_RELEASE}
-
-ARG UBUNTU_RELEASE=xenial
+FROM ${ARCH}/${DISTRO_NAME}:${DISTRO_NICK}
+ARG AT_VERSION
+ARG DISTRO_NAME
+ARG DISTRO_NICK
+ARG EXTRA
+ARG REPO
 
 WORKDIR /root
 
-ENV DEBIAN_FRONTEND=noninteractive
+ENV DEBIAN_FRONTEND noninteractive
+ARG VERSION=${AT_VERSION}${EXTRA}
 
 RUN [ -n "$HTTP_PROXY" ] && echo Acquire::http::proxy '"'$HTTP_PROXY'";' >> /etc/apt/apt.conf.d/http_proxy || :
 RUN [ -n "$HTTPS_PROXY" ] && echo Acquire::https::proxy '"'$HTTPS_PROXY'";' >> /etc/apt/apt.conf.d/https_proxy || :
 RUN [ -n "$FTP_PROXY" ] && echo Acquire::ftp::proxy '"'$FTP_PROXY'";' >> /etc/apt/apt.conf.d/ftp_proxy || :
 
-# Configure apt to nicely install packages
 RUN apt-get update \
-    && apt-get install -y apt-utils \
-        autoconf automake bison byacc ccache cscope ctags curl flex \
-        elfutils libtool libstdc++-4.8-dev m4 make man-db \
+    && apt-get install -y apt-utils curl gnupg \
     && apt-get clean
 
-ARG AT_VERSION=10.0
-
 ENV http_proxy $HTTP_PROXY
 
-RUN curl http://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu/dists/trusty/6976a827.gpg.key | apt-key add - \
-    && echo "deb http://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu ${UBUNTU_RELEASE} at${AT_VERSION}" >> /etc/apt/sources.list \
+RUN curl -s ${REPO}/${DISTRO_NAME}/dists/${DISTRO_NICK}/6976a827.gpg.key | apt-key add - \
+    && echo "deb ${REPO}/${DISTRO_NAME} ${DISTRO_NICK} at${AT_VERSION}" >> /etc/apt/sources.list \
     && apt-get update \
-    && apt-get install -y advance-toolchain-at${AT_VERSION}-runtime advance-toolchain-at${AT_VERSION}-devel \
-        advance-toolchain-at${AT_VERSION}-perf advance-toolchain-at${AT_VERSION}-mcore-libs
+    && apt-get install -y advance-toolchain-at${VERSION}-runtime \
+        advance-toolchain-at${VERSION}-mcore-libs \
+        advance-toolchain-at${VERSION}-devel \
+        advance-toolchain-at${VERSION}-perf
 
-ENV PATH /opt/at${AT_VERSION}/bin:$PATH
+ENV PATH /opt/at${VERSION}/bin:$PATH
```

Comment:
I don't think we can use the same variable for both the path and the package name.
They're slightly different, e.g. for AT 15.0-0 Beta 1, the packages are named `advance-toolchain-at15.0-beta1-runtime` while the directory is `/opt/at15.0-0-beta1/`

Reply:
Right!
I mixed up packages and directories names.
Fixed.